src/framework/musesampler/internal/musesamplerconfiguration.cpp

```diff
@@ -23,27 +23,48 @@
 #include "musesamplerconfiguration.h"
 
 #include <cstdlib>
+#include <QStandardPaths>
 
 using namespace mu;
 using namespace mu::musesampler;
 
 #if defined(Q_OS_LINUX)
-static const io::path_t DEFAULT_PATH("libMuseSamplerCoreLib.so");
+static const io::path_t DEFAULT_LIB_NAME("libMuseSamplerCoreLib.so");
+static io::path_t DEFAULT_PATH()
+{
+    return QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + "/MuseSampler/lib/" + DEFAULT_LIB_NAME;
+}
 #elif defined(Q_OS_MAC)
-static const io::path_t DEFAULT_PATH("/usr/local/lib/libMuseSamplerCoreLib.dylib");
+static const io::path_t DEFAULT_LIB_NAME("libMuseSamplerCoreLib.dylib");
+static io::path_t DEFAULT_PATH()
+{
+    return QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation) + "/MuseSampler/lib/" + DEFAULT_LIB_NAME;
+}
 #else
-static const io::path_t DEFAULT_PATH("MuseSamplerCoreLib.dll");
+static const io::path_t DEFAULT_LIB_NAME("MuseSamplerCoreLib.dll");
+static io::path_t DEFAULT_PATH()
```

Comment:
The three `DEFAULT_PATH` functions could probably be combined with `Qt` logic for path handling.  I wasn't familiar with this, but could look into and change if desired.

Also, I was able to build and test mac + linux locally, but not Windows; I don't have a Windows built setup for MuseScore atm.